src/decrypt.c

```diff
@@ -1,29 +1,36 @@
 #include "ece/keys.h"
 #include "ece/trailer.h"
 
+#include <assert.h>
 #include <string.h>
 
 #include <openssl/evp.h>
 #include <openssl/rand.h>
 
-typedef size_t (*max_decrypted_length_t)(uint32_t rs, size_t ciphertextLen);
+typedef int (*unpad_t)(uint8_t* block, bool lastRecord, size_t* blockLen);
 
-typedef int (*unpad_t)(uint8_t* block, bool isLastRecord, size_t* blockLen);
-
-// Returns the maximum decrypted length of an "aes128gcm" ciphertext.
+// Calculates the maximum plaintext length, including room for the padding
+// delimiter and padding.
 static inline size_t
-ece_aes128gcm_max_decrypted_length(uint32_t rs, size_t ciphertextLen) {
-  size_t numRecords = (ciphertextLen / rs) + 1;
+ece_plaintext_max_length(uint32_t rs, size_t padSize, size_t ciphertextLen) {
+  size_t overhead = padSize + ECE_TAG_LENGTH;
+  if (rs <= overhead) {
+    return 0;
+  }
+  size_t numRecords = ciphertextLen / rs;
+  if (ciphertextLen % rs) {
+    // A smaller final record is expected for "aes128gcm", and required for
```

Comment:
Updated the comment. The final record is smaller than the `rs`, but it should still have an auth tag.